### PR TITLE
docs(typography): docs site to storybook

### PR DESCRIPTION
**Pull request by rise-erpelding**
## Description

This PR continues with migrating the documentation of components from the old docs site to Storybook, migrating content from all docs site typography pages, including the [typography base page](https://opensource.adobe.com/spectrum-css/typography.html), [internationalized typography page](https://opensource.adobe.com/spectrum-css/typography-international.html), [typography body page](https://opensource.adobe.com/spectrum-css/typography-body.html), [typography code page](https://opensource.adobe.com/spectrum-css/typography-code.html), [typography detail page](https://opensource.adobe.com/spectrum-css/typography-detail.html), and [typography heading page](https://opensource.adobe.com/spectrum-css/typography-heading.html).

Here's a general summary of how the docs site Typography pages are organized, and where this is located in the Typography Storybook Docs page:
| Docs Site         | Storybook          |
| ----------------- | ------------------ |
| Typography general page (`/typography.html`): Variants for Heading, Body, Detail, Code | Replaced by Default story showing all Typography components    |
| Typography general page (`/typography.html`): section on applying margins  | Added to description of Default story |
| Typography general page (`/typography.html`): section on migrating from deprecated typography | These changes [are ~4 years old](https://github.com/adobe/spectrum-css/commit/1a604c41a1706d748189f5ed852342f70755ff14#diff-2d99864ab0a927e68ce7f00903f61229cc537e7c8c34c8493ebacecc87455698) so they've been eliminated from the Storybook Docs, but I've moved them to the Changelog for now, happy to hear any opinions that this is unnecessary--I can easily drop this change! |
| Typography heading page (`/typography-heading.html`): Variants, Heavy, and Light sections | All displayed in the two stories for heading variants and heading sizes in the Heading section |
| Typography heading page (`/typography-heading.html`): Typography section showing content hierarchies | Displayed in Heading vs. Body Hierarchy section below the Body stories |
| Typography body page (`/typography-body.html`): Variants section | All displayed in the two stories for body variants and body sizes in the Body section |
| Typography body page (`/typography-body.html`): Standard section showing content hierarchies | Eliminated/already displayed in Heading vs. Body Hierarchy section below the Body stories |
| Typography detail page (`/typography-detail.html`): Variants | All displayed in the two stories for detail variants and detail sizes in the Detail section |
| Typography code page (`/typography-code.html`): Variants | All displayed in the two stories for code variants and code sizes in the Code section, note that the template doesn't support code wrapped in `<pre>` tags, so this is no longer demonstrated (it's not a special feature of the implementation however) but is noted in the story description |
| Internationalized typography page (`/typography-international.html`): Variants for heading, heavy heading, and light heading | Contained in Internationalized Heading sub-section in Internationalized Typography section - sizing has been eliminated |
| Internationalized typography page (`/typography-international.html`): Variants for body | Contained in Internationalized Body sub-section in Internationalized Typography section - sizing has been eliminated |
| Internationalized typography page (`/typography-international.html`): Variants for detail | Contained in Internationalized Detail sub-section in Internationalized Typography section - sizing has been eliminated |
| Internationalized typography page (`/typography-international.html`): Variants for Han | Contained in Internationalized Heading vs. Body Hierarchy sub-section in Internationalized Typography section, below Internationalized Body sub-section |
| Internationalized typography page (`/typography-international.html`): Variants for code | Contained in Internationalized Code sub-section in Internationalized Typography section - sizing and code wrapped with `<pre>` tags have been eliminated |

[CSS-794](https://jira.corp.adobe.com/browse/CSS-794)

## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

### Regression testing

Validate:

1. The documentation pages for at least two other components are still loading, including:

- [ ] The pages render correctly, are accessible, and are responsive.

2. If components have been modified, VRTs have been run on this branch:

- [ ] VRTs have been run and looked at.
- [ ] Any VRT changes have been accepted (by reviewer and/or PR author), or there are no changes.

## To-do list

- [ ] I have read the [contribution guidelines](/.github/CONTRIBUTING.md).
- [ ] I have updated relevant storybook stories and templates.
- [ ] I have tested these changes in Windows High Contrast mode.
- [ ] If my change impacts **other components**, I have tested to make sure they don't break.
- [ ] If my change impacts **documentation**, I have updated the documentation accordingly.
- [ ] ✨ This pull request is ready to merge. ✨

### CODE REVIEW

File: components/typography/stories/typography.stories.js [hunk 1]
```diff
@@ -115,57 +131,185 @@ WithForcedColors.parameters = {
...
 };
-Body.parameters = {
+BodySizes.parameters = {
+	chromatic: { disableSnapshot: true },
+};
+
+/**
+ * When typography elements are paired, such as with heading and body below, clear content hierarchies are shown.
+ * 
+ * Note that the body component is not available in XXS, so pairing with the size XXS heading is not displayed here.
```
Comment (jawinn) [inline components/typography/stories/typography.stories.js:175]: On the docs site, I see that an XXS heading is also shown paired with XS body text. Is that fine to no longer show? If we're not showing it, would rewording this sentence to something like this be helpful: "Since the body component is not available in size XXS, the size XXS heading can be paired with the size XS body."

File: components/typography/stories/typography.stories.js [hunk 2]
```diff
@@ -115,57 +131,185 @@ WithForcedColors.parameters = {
...
-Body.argTypes = {
+/**
+ * Body is primarily used for Spectrum components and for blocks of text.
+ */
+export const BodySizes = (args, context) => Sizes({
+	Template,
+	direction: "column",
+	...args,
+}, context);
+BodySizes.argTypes = {
```
[the remark this answers is not in the document]
Comment (rise-erpelding) [reply components/typography/stories/typography.stories.js:158]: Totally silly but consistent at least!

File: components/typography/stories/template.js [hunk 3]
```diff
@@ -106,7 +112,413 @@ export const Template = (args = {}, context = {}) => {
 	/** Wrap items with the spectrum-Typography wrapper if there are more than 1 items (this ensures correct margins) */
 	return html`${when(
 		content.length > 1,
-		() => html`<div class="spectrum-Typography" id=${ifDefined(id)}>${processedContent}</div>`,
+		() => html`<div class="spectrum-Typography" id=${ifDefined(id)} lang=${lang || nothing}>${processedContent}</div>`,
```
Comment (rise-erpelding) [inline components/typography/stories/template.js:115]: This was the most minimal way I could think of to add the lang attribute if it's passed

File: components/typography/stories/template.js [hunk 4]
```diff
@@ -106,7 +112,413 @@ export const Template = (args = {}, context = {}) => {
...
+	content: html`
+	${Container({
+		direction: "column",
+		heading: "Japanese",
+		wrapperStyles: {"row-gap": "0"},
+		content: html`
+				${Template({
+					...args,
+					context,
+					content: ["暗号", "Code"],
```
Comment (rise-erpelding) [inline components/typography/stories/template.js:494]: The lang attribute is set in `Template` to be added to the outer `.spectrum-Typography` `<div>` wrapper, but this wrapper only works if there's more than 1 item--this is a little bit hacky but this is the only place where it needs to be used.

File: components/typography/stories/template.js [hunk 5]
```diff
@@ -129,6 +130,7 @@ const RegularBoldItalicGroup = (args, context) => Container({
 			content: [
 				{
 					content: `Regular ${args.semantics} text`,
+					customStyles: {"margin-block-start": "0"},
```
Comment (rise-erpelding) [inline components/typography/stories/template.js:133]: Some spacing updates also got added in with this lang attribute commit! I didn't love the uneven/uncollapsed spacing before headings, but can leave these custom styles off if there's a good reason to do so!

File: components/typography/stories/template.js (hunk 3 above)
Comment (jawinn) [reply components/typography/stories/template.js:115]: Nice, I didn't know `nothing` would also work like that with attributes. `lang=${ifDefined(lang)}` might also work here if it doesn't need to also check against an empty string.

**Review by jawinn — APPROVED** [review]
Looks good to me! I see those requested changes addressed, and the new `lang` attribute appearing.

File: components/typography/stories/template.js [hunk 6]
```diff
@@ -21,6 +21,7 @@ export const Template = (args = {}, context = {}) => {
 		customClasses = [],
 		customStyles = {},
 		skipLineBreak = false,
+		lang = "",
```
Comment (castastrophe) [inline components/typography/stories/template.js:24]: You can fetch the global lang from the context object like this and use it as a fallback if you like for when the block doesn't define it's own lang property.

```
	const { globals = {} } = context;
	const lang = globals.lang ?? "en-US";
```

File: components/typography/stories/template.js (hunk 3 above)
Comment (castastrophe) [inline components/typography/stories/template.js:115]: The ifDefined syntax would be the best practice here or you could use the fallback of the global lang value:

```suggestion
		() => html`<div class="spectrum-Typography" id=${ifDefined(id)} lang=${ifDefined(lang)}>${processedContent}</div>`,
```

File: components/typography/stories/template.js [hunk 7]
```diff
@@ -106,7 +112,413 @@ export const Template = (args = {}, context = {}) => {
...
+	emphasized: "גוף חזק",
+	strong: "גוף חזק",
+};
+
+export const DocsInternationalizedHeadingVariants = (args, context) => Container({
+	withBorder: false,
+	content: html`
+		${Container({
+			direction: "column",
+			heading: "Japanese (Default)",
```
Comment (castastrophe) [inline components/typography/stories/template.js:313]:
```suggestion
			heading: "Japanese (default)",
```

I believe this format would be sentence case per the style guide. Would you mind double checking the headings?

**Review by castastrophe — APPROVED** [review]
This looks really great - nice approach to a very complex migration!

File: components/typography/stories/template.js (hunk 7 above)
Comment (rise-erpelding) [reply components/typography/stories/template.js:313]: Great catch, thanks!

File: components/typography/stories/template.js [hunk 8]
```diff
@@ -21,6 +21,7 @@ export const Template = (args = {}, context = {}) => {
 		customClasses = [],
 		customStyles = {},
 		skipLineBreak = false,
+		lang = context.globals?.lang || "en-US",
```
Comment (rise-erpelding) [inline components/typography/stories/template.js:24]: @castastrophe What do you think of this approach? The `lang` can still be passed in as an arg so that I can set multiple different `lang`s within a story, but if it's not passed in we'll try to find it in the global context or default to English.

Comment (castastrophe) [reply components/typography/stories/template.js:24]: I love it!  Maximum flexibility.

**Review by castastrophe — APPROVED** [review]
